### PR TITLE
Build a universal binary based on syncthing 1.13.0-rc.1

### PR DESCRIPTION
**Pull request by fredoliveira**
This PR tweaks a number of configuration options in order to grab and build a universal binary build of `syncthing-macos`, bundling the universal build of `syncthing`, which at the time of this writing, is at version `1.13.0-rc.1`. In short, the changes proposed by this PR are:

- An upgrade to sparkle 1.24, which supports M1 processor-based macs
- A change to the script that grabs `syncthing` binaries for bundling
- Some small tweaks to handle deprecations and other issues pointed by Xcode

Fixes syncthing/syncthing-macos#138 (see the issue itself for further discussion)

### CODE REVIEW

File: syncthing/Info.plist [hunk 1]
```diff
@@ -19,7 +19,7 @@
 	<key>CFBundlePackageType</key>
 	<string>APPL</string>
 	<key>CFBundleShortVersionString</key>
-	<string>1.12.0-1</string>
+	<string>1.13.0-rc1</string>
 	<key>CFBundleVersion</key>
 	<string>101200001</string>
```
Comment (xor-gate) [inline syncthing/Info.plist:24]: You also need to update the CFBundleVersion prefix 1012 to 1013

**Comment by xor-gate** [comment]
You are able to grab the dmg at https://build.syncthing.net/viewLog.html?buildId=91010&buildTypeId=SyncthingMacOS_BuildReleaseDmg&tab=artifacts also from github Show all checks -> Build (Release DMG) (Syncthing macOS) -> Teamcity build artifacts.

**Comment by xor-gate** [comment]
@fredoliveira probably we should also target older SDK versions for better compatibility

@calmh it seems we are in trouble again with the notarized stuff for this PR dmg https://build.syncthing.net/repository/download/SyncthingMacOS_BuildReleaseDmg/91010:id/Syncthing-1.13.0-rc1.dmg

Screenshot 2021-01-15 at 11 01 09

Screenshot 2021-01-15 at 11 12 22

I used Taccy for this https://eclecticlight.co/taccy-signet-precize-alifix-utiutility-alisma/

alltrough:
```
codesign --verify --verbose --deep /Applications/Syncthing.app
/Applications/Syncthing.app: valid on disk
/Applications/Syncthing.app: satisfies its Designated Requirement
```
```
spctl --assess -vv /Applications/Syncthing.app
/Applications/Syncthing.app: accepted
source=Developer ID
origin=Developer ID Application: Jakob Borg (LQE5SYM783)
```

```
codesign -d --deep --verbose=2 -r- /Applications/Syncthing.app
Executable=/Applications/Syncthing.app/Contents/MacOS/Syncthing
Identifier=com.github.xor-gate.syncthing-macosx
Format=app bundle with Mach-O universal (x86_64 arm64)
CodeDirectory v=20500 size=2096 flags=0x10000(runtime) hashes=58+3 location=embedded
Signature size=9000
Authority=Developer ID Application: Jakob Borg (LQE5SYM783)
Authority=Developer ID Certification Authority
Authority=Apple Root CA
Timestamp=14 Jan 2021 at 22:23:19
Info.plist entries=32
TeamIdentifier=LQE5SYM783
Runtime Version=11.1.0
Sealed Resources version=2 rules=13 files=46
Nested=Frameworks/libswiftIOKit.dylib
Nested=Frameworks/libswiftXPC.dylib
Nested=Frameworks/libswiftObjectiveC.dylib
Nested=Frameworks/Sparkle.framework
Nested=Frameworks/libswiftCore.dylib
Nested=Frameworks/libswiftDispatch.dylib
Nested=Frameworks/libswiftFoundation.dylib
Nested=Frameworks/libswiftCoreGraphics.dylib
Nested=Frameworks/libswiftDarwin.dylib
Nested=Frameworks/libswiftCoreFoundation.dylib
designated => identifier "com.github.xor-gate.syncthing-macosx" and anchor apple generic and certificate 1[field.1.2.840.113635.100.6.2.6] /* exists */ and certificate leaf[field.1.2.840.113635.100.6.1.13] /* exists */ and certificate leaf[subject.OU] = LQE5SYM783
```

**Comment by xor-gate** [comment]
I have bumped v1.12.1 also in develop, it seems the macos builder for release doesn't notarize anymore also for the v1.12.1: https://build.syncthing.net/viewLog.html?buildId=91088&buildTypeId=SyncthingMacOS_BuildReleaseDmg&tab=artifacts

**Comment by xor-gate** [comment]
Sorry for the noise @calmh , but the Taccy application cached the loading of the app. It seems [v1.12.1](https://build.syncthing.net/viewLog.html?buildId=91088&buildTypeId=SyncthingMacOS_BuildReleaseDmg&tab=artifacts) is correctly notarized.

**Comment by calmh** [comment]
You need to notarize the thing before attempting to run it or you'll get the dialogs shown.

**Comment by xor-gate** [comment]
You are right it seems the PR doesn't get notarized, will it only be triggered on the develop branch? https://build.syncthing.net/viewType.html?buildTypeId=Release_NotarizeSyncthingMacosMac&branch_Release=issue-138&tab=buildTypeStatusDiv

**Pull request merged**